kcwitools/optimal_extract_utility.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import print_function
+
+import numpy as np
+import matplotlib.pyplot as plt
+from numpy.polynomial.chebyshev import chebfit,chebval
+
+
+
+# Adopted from some online stuff that uses these for optimal extraction
+
+def wavelegth_fitter( fluxcube, varcube, aperture_mask, wavelength, order, aperture_indices, plot=False, verbose=False):
+
+	"""
+	Fit polynomials in the wavelength direction of each pixel in an aperture. 
+	
+	Input: 
+		 fluxcube 			:    KCWI flux cube 
+		 varceub  			:	 KCWI variance cube
+		 aperture_mask  	:    3D mask for the pseudo-slit. Good values=True, Bad ones =False
+		 wavelength     	:    wavelength grid of the fluxcube
+		 order          	:    Chebyshev polynomial order for fitting
+		 aperture_indices 	:    Indices of the cube corresponding to pseudo-slit aperture, in an Nx2 array. 
+
+		 plot [False]		:  If true plots the polynomials (both before and after normalising spatially)    
+		 verbose [False]    :  True prints stuff in terminal
+
+
+	"""
+
+
+	wavedim,ydim,xdim=fluxcube.shape
+	wavelength_fits=np.zeros([wavedim, ydim, xdim])
+
+
+	#Lamdas must be scaled to between -1 and 1 for Chebyshev fitting
+	min_val=wavelength.min()
+	max_val=wavelength.max()
+	scaled_l=np.array([2*(lamda-min_val)/(max_val-min_val) -1 for lamda in wavelength])
```

Comment:
We have pretty good code for fiitting in PypeIt.
Maybe we should make that a dependency for `kcwitools`
and use them..

Reply:
@profxj : I am happy to do that. Could you help rewrite this routine?